src/main/java/frc/robot/leds/LedStrip.java

```diff
@@ -0,0 +1,181 @@
+        ledBuffer = new AddressableLEDBuffer(length);
+        ledStrip.setLength(length);
+        stripLength = length;
+    }
+
+    public void setState(LedState state){
+        setBlinkTime(state.getBlinkTime());
+        setFadeDuration(state.getFadeDuration());
+        setMode(state.getMode());
+        setPrimary(state.getPrimary());
+        setSecondary(state.getSecondary());
+    }
+
+    /**
+     * Sets the primary color of the LEDs.
+     * It will be used when in the solid mode and as the
+     * initial color in the blink and fade modes
+     * @param color
+     */
+    private void setPrimary(Color color) {
+        primary = color;
+    }
+
+    /**
+     * Sets the secondary color of the LEDs.
+     * It will be used as the second color in the blink and fade modes
+     * @param color
+     */
+    private void setSecondary(Color color) {
+        secondary = color;
+    }
+
+    private void setMode(LedMode mode) {
+        this.mode = mode;
+    }
+
+    /**
+     * Sets the offset of the LEDs.
+     * @param offset The amount of LEDs that won't light.
+     *               Counts from the start of the strip.
+     */
+    public void setOffset(int offset){
+        setSolidColor(Color.kBlack, 0, offset+1);
+        this.offset = offset;
+    }
+
+    public void setPercentage(int percentage) {
+        this.percentage = percentage;
+    }
+
+    /**
+     * Sets the led blink time.
+     * @param blinkTime Blink time in seconds
+     */
+    private void setBlinkTime(double blinkTime) {
+        this.blinkTime = blinkTime;
+    }
+
+    /**
+     * Sets the duration of the fade effect.
+     * @param duration Fade duration in seconds.
+     */
+    private void setFadeDuration(double duration) {
+        fadeDuration = duration;
+    }
+
+    private void setSolidColor(Color color) {
+        var appliedColor = new Color(color.red, color.green, color.blue);
+        for (int i = offset; i < stripLength; i++) {
+            ledBuffer.setLED(i, appliedColor);
+        }
+        ledStrip.setData(ledBuffer);
+    }
+
+    private void setSolidColor(Color color, int start, int end) {
+        var appliedColor = new Color(color.red, color.green, color.blue);
+        for (int i = start; i < end; i++) {
+            ledBuffer.setLED(i, appliedColor);
+        }
+        ledStrip.setData(ledBuffer);
+    }
+
+    /**
+     * Updates fadeColor to the correct color for a fade effect
+     * at the current time by interpolating the two given colors.
+     * @param initial Initial color.
+     * @param goal    Final Color.
+     */
+    private void updateFade(Color initial, Color goal) {
+        Translation3d initialPoint = new Translation3d(initial.red, initial.green, initial.blue);
+        Translation3d goalPoint = new Translation3d(goal.red, goal.green, goal.blue);
+
+        double d = initialPoint.getDistance(goalPoint) / fadeDuration;
+        double t = d / (initialPoint.minus(goalPoint)).getNorm();
```

Comment:
There is rarely a need to shorten names. In this case it's not clear what their meaning is.